### PR TITLE
SF-2018: Filter out undefined query parameters

### PR DESCRIPTION
**Pull request by rklind**

### CODE REVIEW

File: src/sdk/http/http-client.ts [hunk 1]
```diff
@@ -99,12 +99,22 @@ export abstract class HttpClient<TRequest, TResponse> implements IHttpClient {
 		}
 	}
 
+	static getFilteredParameters(params?: any): any {
+		let filteredParams = {};
```
Comment (smbecker) [inline src/sdk/http/http-client.ts]: `let filteredParams;`

File: src/sdk/http/http-client.ts [hunk 2]
```diff
@@ -99,12 +99,22 @@ export abstract class HttpClient<TRequest, TResponse> implements IHttpClient {
...
+		for (const param in params) {
+			if (params[param] !== undefined) {
```
Comment (smbecker) [inline src/sdk/http/http-client.ts]: Ensure `filteredParams` is defined here. `if (!filteredParams) { filteredParams = {}; }`

File: src/sdk/http/http-client.ts [hunk 3]
```diff
@@ -99,12 +99,22 @@ export abstract class HttpClient<TRequest, TResponse> implements IHttpClient {
...
+			}
+		}
+		return filteredParams;
+	}
+
 	static getUrl(requestPath: string, baseUrl?: string, params?: any): string {
 		if (baseUrl && baseUrl.length > 0 && requestPath.indexOf('://') === -1) {
 			requestPath = baseUrl + requestPath;
 		}
 		if (params) {
```
Comment (smbecker) [inline src/sdk/http/http-client.ts]: Change this to...

```
if (params) {
	params = HttpClient.getFilteredParameters(params);
}
if (params) {
	//qs.stringify
}
```

File: src/sdk/http/http-client.ts [hunk 4]
```diff
@@ -99,12 +99,22 @@ export abstract class HttpClient<TRequest, TResponse> implements IHttpClient {
 		}
 	}
 
+	static getFilteredParameters(params?: any): any {
```
Comment (smbecker) [inline src/sdk/http/http-client.ts]: I don't think this needs to be a `static` method attached to `HttpClient`. I would move it up so it isn't a part of the public api.

Comment (rklind) [reply src/sdk/http/http-client.ts]: will do

File: src/sdk/http/http-client.ts (hunk 3 above)
Comment (rklind) [reply src/sdk/http/http-client.ts]: will do

File: src/sdk/http/http-client.ts (hunk 2 above)
Comment (rklind) [reply src/sdk/http/http-client.ts]: will do

File: src/sdk/http/http-client.ts (hunk 1 above)
Comment (rklind) [reply src/sdk/http/http-client.ts]: will do

File: src/sdk/http/http-client.ts [hunk 5]
```diff
@@ -69,6 +69,19 @@ function invalidTokenError() {
 	return new Error('Could not acquire a valid access token.');
 }
 
+function getFilteredParameters(params?: any): any {
```
Comment (jtirjan) [inline src/sdk/http/http-client.ts]: Method name isn't very descriptive. I'd suggest something more like `removeUndefinedParameters()`

Comment (rklind) [reply src/sdk/http/http-client.ts]: will do, thanks!

**Pull request merged**